src/pages/docs/guides/chat/build-task-oriented.mdx

```diff
@@ -0,0 +1,419 @@
+const currentTime = Math.round(Date.now() / 1000);
+
+// The clientId could be the users ID in your application,
+// or a random string if you want to make users anonymous.
+//
+// The capabilities here allow the holder of this token to publish and subscribe to messages,
+// typing indicators and room reactions in the chat room called "foo".
+
+const claims = {
+    "iat": currentTime, /* current time in seconds */
+    "exp": currentTime + 14400, /* time of expiration in seconds */
+    "x-ably-capability": "{\"foo\":[\"publish\", \"subscribe\"]}",
+    "x-ably-clientId": "your-client-id",
+}
+
+ablyJwt = jwt.sign(
+    claims,
+    "{{ API_KEY_SECRET }}",
+    { header: header }
+)
+
+console.log('JWT is: ' + ablyJwt);
+```
+</Code>
+
+How you authenticate is also key. To balance security and experience, you want short-lived tokens that can be easily revoked if a user is misbehaving or needs their permissions changed, but automatically expire after a period of time. This means that if a token is compromised, it will only be valid for a limited time. **In production apps, you should not use API keys for client-side authentication**. You can use them server-side, but as they are long-lived and require explicit revocation, exposure to untrusted users poses a continuing risk.
+
+Authentication is best achieved using JSON Web Tokens (JWTs). These are tied to a particular `clientId` and come with a set of [capabilities](/docs/chat/setup#authentication) that control what a client can and cannot do - for example whether they can send messages, join a certain room or moderate. Ably's SDKs handle the timing and process of requesting a new token for you, refreshing it when it expires. All you need to do is provide a server-side endpoint that can generate the JWT for the client. This enables clients to use your existing authentication systems or user sessions to generate their Ably token.
+
+## Presence: Know who's available in the room
+
+[Presence](/docs/chat/rooms/presence) shows you who's currently active in the room. This is especially valuable for scenarios where task completion depends on participant availability:
+
+* **Support tickets:** See when agents are online and available to help.
+* **Deliveries:** Know if the delivery driver or customer is actively monitoring the chat.
+* **Gaming sessions:** See which players are currently active in the game.
+* **Collaborative work:** Know who's available for real-time discussion.
+
+Beyond just online/offline status, presence can include rich information:
+
+* **Current status:** "Available", "In a meeting", "Driving", "On break".
+* **Task context:** "Working on ticket #123", "In delivery zone A".
+* **User info:** Use presence to show an avatar, display name, role, or other information about the user.
+
+This contextual presence information helps participants understand not just who's online, but who's ready and able to engage with the current task.
+
+```javascript
+// subscribe to presence events
```

Comment:
Does this need an `attach` call after it to make the example complete?

Reply:
added attach calls in examples where it makes sense